src/main/java/graphql/schema/DataFetcher.java

```diff
@@ -3,5 +3,5 @@
 
 public interface DataFetcher {
```

Comment:
Generics. I'm looking forward to that one too. No special reason. Like the thought of adding type safety as a feature. It has the potential for users to reduce code using generics. Doesn't sound like a breaking change though. Worst case scenario is that someone using the non typed version gets compilation warnings. I think.

Agree though that the scope of this PR should be limited to the clean up of what a data fetcher does.